### PR TITLE
Vanilla Factions Expanded - Settlers

### PR DESCRIPTION
**Pull request by onigamii**
## Additions

Patches the Settlers mod from Vanilla Factions Expanded

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long)

### CODE REVIEW

File: Patches/Vanilla Factions Expanded - Settlers/Drugs/Alcohol_Chemshine.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            </statOffsets>
+          </value>
+        </li>
+
+        <!-- === Addictivines === -->
+        <li Class="PatchOperationReplace">
+          <xpath>/Defs/ThingDef[defName="Chemshine"]/comps/li[@Class="CompProperties_Drug"]/addictiveness</xpath>
+
+          <value>
+            <addictiveness>0.05</addictiveness>
```
Comment (MadaraUchiha) [inline Patches/Vanilla Factions Expanded - Settlers/Drugs/Alcohol_Chemshine.xml:36]: Why this change? It doesn't seem CE related.

File: Patches/Vanilla Factions Expanded - Settlers/Drugs/Alcohol_Chemshine.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          </value>
+        </li>
+
+        <li Class="PatchOperationReplace">
+          <xpath>/Defs/ThingDef[defName="Chemshine"]/comps/li[@Class="CompProperties_Drug"]/minToleranceToAddict</xpath>
+
+          <value>
+            <minToleranceToAddict>0.05</minToleranceToAddict>
```
Comment (MadaraUchiha) [inline Patches/Vanilla Factions Expanded - Settlers/Drugs/Alcohol_Chemshine.xml:44]: Why this change? it doesn't seem CE related.

File: Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Bandits.xml [hunk 3]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+
+  <Operation Class="PatchOperationFindMod">
+    <mods>
+      <li>Vanilla Factions Expanded - Settlers</li>
+    </mods>
+
+    <match Class="PatchOperationSequence">
+      <success>Always</success>
```
Comment (MadaraUchiha) [inline Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Bandits.xml:10]: Is the `<success>Always</success>` required here? I tend to prefer not hiding errors when they occur.

File: Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+
+  <Operation Class="PatchOperationFindMod">
+    <mods>
+      <li>Combat Extended</li>
```
Comment (MadaraUchiha) [inline Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml:6]: This *is* the Combat Extended mod, so this condition will always match. Did you mean to search for Vanilla Factions Expanded - Settlers?

```suggestion
      <li>Vanilla Factions Expanded - Settlers</li>
```

File: Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    </mods>
+
+    <match Class="PatchOperationSequence">
+      <success>Always</success>
```
Comment (MadaraUchiha) [inline Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml:10]: Is the `<success>Always</success>` required here? I tend to prefer not hiding errors when they occur.

File: Patches/Vanilla Factions Expanded - Settlers/Scenarios/Scenarios_Classic.xml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  <Operation Class="PatchOperationFindMod">
+    <mods>
+      <li>Vanilla Factions Expanded - Settlers</li>
+    </mods>
+
+    <match Class="PatchOperationSequence">
+      <operations>
+        <!-- === Dynamine Count === -->
+        <li Class="PatchOperationReplace">
+          <xpath>/Defs/ScenarioDef[defName="VFES_Bandits"]/scenario/parts/li[10]</xpath>
```
Comment (MadaraUchiha) [inline Patches/Vanilla Factions Expanded - Settlers/Scenarios/Scenarios_Classic.xml:13]: Is there a way to do this without resorting to using the `li`'s index? Anything changes the order and the patch might blow up.

File: Patches/Vanilla Factions Expanded - Settlers/ThingDefs_Buildings/TurretGatlingGun.xml [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+          </FireModes>
+        </li>
+
+        <!-- === Turret === -->
+        <!-- == thingClass == -->
+        <li Class="PatchOperationReplace">
+          <xpath>/Defs/ThingDef[defName="Turret_GatlingGun"]/thingClass</xpath>
+
+          <value>
+            <thingClass>CombatExtended.Building_TurretGunCE</thingClass>
```
Comment (MadaraUchiha) [inline Patches/Vanilla Factions Expanded - Settlers/ThingDefs_Buildings/TurretGatlingGun.xml:57]: Not adding ammo to it (I might have missed that)

File: Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml (hunk 4 above)
Comment (onigamii) [reply Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml:6]: Completely missed this lmao, thanks

File: Patches/Vanilla Factions Expanded - Settlers/ThingDefs_Buildings/TurretGatlingGun.xml (hunk 7 above)
Comment (onigamii) [reply Patches/Vanilla Factions Expanded - Settlers/ThingDefs_Buildings/TurretGatlingGun.xml:57]: Dunno why this is missing, they of course have ammo as i've been using them for almost a year in game. Something must've messed up because the branch is indeed up to date with my local changes.

File: Patches/Vanilla Factions Expanded - Settlers/Scenarios/Scenarios_Classic.xml (hunk 6 above)
Comment (onigamii) [reply Patches/Vanilla Factions Expanded - Settlers/Scenarios/Scenarios_Classic.xml:13]: We're on the same boat, i tried a couple of options but this was the only one that actually worked, just told myself that if it breaks i'll find what's breaking it and fixing it.

Comment (MadaraUchiha) [reply Patches/Vanilla Factions Expanded - Settlers/Scenarios/Scenarios_Classic.xml:13]:
```suggestion
          <xpath>/Defs/ScenarioDef[defName="VFES_Bandits"]/scenario/parts/li[thingDef="VFES_Weapon_GrenadeDynamite"]</xpath>
```

How about this one? ^

File: Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml (hunk 5 above)
Comment (MadaraUchiha) [reply Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Settlers.xml:10]: image

File: Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Bandits.xml (hunk 3 above)
Comment (MadaraUchiha) [reply Patches/Vanilla Factions Expanded - Settlers/PawnKindDefs/PawnKinds_Bandits.xml:10]: image

File: Patches/Vanilla Factions Expanded - Settlers/Scenarios/Scenarios_Classic.xml (hunk 6 above)
Comment (onigamii) [reply Patches/Vanilla Factions Expanded - Settlers/Scenarios/Scenarios_Classic.xml:13]: Done
